app/Platform/Actions/UpdateGameMetricsAction.php

```diff
@@ -4,47 +4,106 @@
+            ->count();
+        $game->players_hardcore = $game->playerGames()
+            ->leftJoin('UserAccounts as user', 'user.ID', '=', 'player_games.user_id')
+            ->where('player_games.achievements_unlocked_hardcore', '>', 0)
+            ->where('user.Untracked', false)
+            ->count();
+
+        $achievementSetVersionChanged = false;
+        $achievementsPublishedChange = 0;
+        $pointsTotalChange = 0;
+        $playersTotalChange = 0;
+        $playersHardcoreChange = 0;
+        if ($game->achievements_published || $game->achievements_unpublished) {
+            $achievementSetVersionHashPayload = $game->achievements()->published()
+                ->orderBy('ID')
+                ->get(['ID', 'MemAddr', 'Points'])
+                ->map(fn ($achievement) => implode('-', $achievement->only(['ID', 'MemAddr', 'Points'])))
+                ->implode('-');
+            $game->achievement_set_version_hash = hash('sha256', $achievementSetVersionHashPayload);
+
+            $achievementSetVersionChanged = $game->isDirty('achievement_set_version_hash');
+            $achievementsPublishedChange = $game->achievements_published - $game->getOriginal('achievements_published');
+            $pointsTotalChange = $game->points_total - $game->getOriginal('points_total');
+            $playersTotalChange = $game->players_total - $game->getOriginal('players_total');
+            $playersHardcoreChange = $game->players_hardcore - $game->getOriginal('players_hardcore');
+        }
+        $pointsWeightedBeforeUpdate = $game->TotalTruePoints;
+
+        $game->save();
+
+        // dispatch(new UpdateGameAchievementsMetricsJob($game->id))->onQueue('game-metrics');
+        app()->make(UpdateGameAchievementsMetricsAction::class)
+            ->execute($game);
+        $game->refresh();
+        $pointsWeightedChange = $game->TotalTruePoints - $pointsWeightedBeforeUpdate;
+
+        GameMetricsUpdated::dispatch($game);
 
-        /*
-         * update unpublished achievements metrics
-         */
-        // $this->attributes['achievements_unpublished'] = $this->achievements()->unpublished()->count();
+        // TODO dispatch events for achievement set and game metrics changes
+        $tmp = $achievementsPublishedChange;
+        $tmp = $pointsTotalChange;
+        $tmp = $pointsWeightedChange;
+        $tmp = $playersTotalChange;
+        $tmp = $playersHardcoreChange;
 
-        // $this->attributes['players_total'] = $this->players()->count();
+        // ad-hoc updates for player games, so they can be updated the next time a player updates their profile
+        // Note that those might be multiple thousand entries depending on a game's players count
 
-        // $game->save();
+        $updateReason = null;
+        if ($pointsWeightedChange) {
+            $updateReason = 'weighted_points_outdated';
+        }
+        if ($achievementSetVersionChanged) {
+            $updateReason = 'version_mismatch';
+        }
```

Comment:
Is it possible that `$pointsWeightedChange` and `$achievementSetVersionChanged` could both be truthy simultaneously?

Reply:
Absolutely, `version_mismatch` would take precedence in that case, which is fine.
Then again, i'm not too happy with that area yet - it was the latest addition to the process and there is no entry path yet where all of the outdated player games get added to the queue.